jbmc/src/java_bytecode/replace_java_nondet.cpp

```diff
@@ -282,8 +281,8 @@ static goto_programt::targett check_and_replace_target(
     inserted_expr.set_nullable(
       instr_info.get_nullable_type() ==
       nondet_instruction_infot::is_nullablet::TRUE);
-    target_instruction->code = code_returnt(inserted_expr);
-    target_instruction->code.add_source_location() =
+    target_instruction->code_nonconst() = code_returnt(inserted_expr);
+    target_instruction->code_nonconst().add_source_location() =
       target_instruction->source_location;
```

Comment:
Why did we ever need to replace the entire `code` - wouldn't the `return_value()` suffice?